jobbergate-api/jobbergate_api/apps/job_scripts/job_script_files.py

```diff
@@ -0,0 +1,269 @@
+"""
+Provide a convenience class for managing job-script files.
+"""
+
+from pathlib import Path
+from typing import Any, Dict, List, cast
+
+from buzz import Buzz, require_condition
+from file_storehouse import FileManager
+from jinja2 import Template
+from loguru import logger
+from pydantic import BaseModel, root_validator
+
+from jobbergate_api.apps.applications.application_files import ApplicationFiles
+from jobbergate_api.apps.applications.schemas import ApplicationConfig
+from jobbergate_api.config import settings
+from jobbergate_api.s3_manager import IO_TRANSFORMATIONS, file_manager_factory, s3_client
+
+JOBSCRIPTS_WORK_DIR = "job-scripts"
+JOBSCRIPTS_MAIN_FILE_FOLDER = "main-file"
+JOBSCRIPTS_SUPPORTING_FILES_FOLDER = "supporting-files"
+
+
+class JobScriptCreationError(Buzz):
+    """Raise exception when facing any error creating a job-script from application files."""
+
+
+def flatten_param_dict(param_dict: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Flatten an input dictionary to support the rendering process.
```

Comment:
Please include an example in the docstring here showing how the dictionary is flattened that shows an example input and the example output.